server/middleware/token.js

```diff
@@ -96,7 +96,7 @@ function token(options) {
     var registry = app.registry;
     if (!TokenModel) {
       if (registry === loopback.registry) {
-        TokenModel = options.model || loopback.AccessToken;
+        TokenModel = app.models[options.model] || loopback.AccessToken;
```

Comment:
Good catch!

I think we need to support the case where `options.model` is a model instance too, see e.g. https://github.com/strongloop/loopback/blob/ed4f56c7f8f8405e0b61246af3a5f9d2382b78da/server/middleware/rest.js#L52-L53

I am proposing to use `loopback.getModel` (`registry.getModel`) instead of `app.models`, because the former is already able to handle both string and constructor arguments. In fact, that's already done in the code below. I think it will be best to simplify this whole code to the following:

```js
if (registry === loopback.registry && !options.model) {
  TokenModel = loopback.AccessToken;
} else {
  TokenModel = registry.getModel(options.model || 'AccessToken');
}
```